djangocms_picture/models.py

```diff
@@ -255,19 +255,23 @@ def get_size(self, width=None, height=None):
             if crop:
                 if not height and width:
                     if self.picture.width > self.picture.height:
-                        height = int(width / PICTURE_RATIO)
+                        height = width / PICTURE_RATIO
                     else:
-                        height = int(width * PICTURE_RATIO)
+                        height = width * PICTURE_RATIO
 
                 elif not width and height:
                     if self.picture.width > self.picture.height:
-                        width = int(height * PICTURE_RATIO)
+                        width = height * PICTURE_RATIO
                     else:
-                        width = int(height / PICTURE_RATIO)
+                        width = height / PICTURE_RATIO
 
             width = width or self.picture.width
             height = height or self.picture.height
 
+        # ensure width and height are int
```

Comment:
I like the approach since it is consistent. 

However, I am not sure that both `width` or `height` could be `None` in certain situations: If `width` and `height` fields are empty , no thumbnail option is given, automatic scaling is `False` and an external picture is selected, AFAICS there might be a `None`. 

Does it make sense to have a default size `(640, 640 / PICTURE_RATIO)` if both `width` and `height` are `None`? If only one is `None` calculate the other by multiplying/dividing by `PICTURE_RATIO`?

Reply:
I don't have an opinion on this question. In any case it is not related to this PR because the float problem comes from the fact that FilerImageField can return float. It it not related with computations done here.

Reply:
Right. My concern is that `int(None)` raises a server error as opposed to lead to just an incorrectly rendered template (such as `<img ... width="None" ...>`).   @marksweb What do you think? We can go over the logic in a second PR. I have a starting point with djangocms-frontend?

Reply:
Ah ok I understand. You are tre. I should add a condition to ensure the `width` or `height` is not `None`:
```
if width:
    width = int(width)
```